packages/skill-api/src/core/services/transfer-purchase.ts

```diff
@@ -163,92 +166,100 @@ export async function transferPurchase({
     // creates a transfer record
     // update merchant records if the purchase involved stripe
 
-    if (purchase?.merchantCharge?.merchantCustomer) {
+    const updates = []
+
+    if (purchase?.merchantCharge) {
       console.log(
-        `stripe customer ${purchase.merchantCharge.merchantCustomer.identifier}`,
+        `update ${purchase.merchantCharge.id} user to ${targetUser.id}`,
       )
-      const {identifier} = purchase.merchantCharge.merchantCustomer
 
-      // try {
-      //   const existingCustomer = (await stripe.customers.retrieve(
-      //     identifier,
-      //   )) as Stripe.Response<Stripe.Customer>
+      const updateMerchantCharge = prisma.merchantCharge.update({
+        where: {
+          id: purchase.merchantCharge.id,
+          userId: sourceUser.id,
+        },
+        data: {
+          userId: targetUser.id,
+        },
+      })
 
-      //   await stripe.customers.update(identifier, {
-      //     email: targetUser.email,
-      //     name: targetUser.name || existingCustomer.name || targetUser.email,
-      //   })
-      // } catch (error) {
-      //   console.error('Error updating stripe customer', error)
-      // }
+      updates.push(updateMerchantCharge)
+    }
 
+    if (purchase?.merchantCharge?.merchantCustomer) {
       console.log(
-        `update ${purchase.merchantCharge.id} user to ${targetUser.id}`,
+        `stripe customer ${purchase.merchantCharge.merchantCustomer.identifier}`,
       )
 
-      // const updateMerchantCharge = prisma.merchantCharge.update({
-      //   where: {
-      //     id: purchase.merchantCharge.id,
-      //   },
-      //   data: {
-      //     userId: targetUser.id,
-      //   },
-      // })
+      const {identifier} = purchase.merchantCharge.merchantCustomer
+      const isCustomer =
+        purchase.merchantCharge.merchantAccount.label !== 'egghead-stripe' &&
+        identifier &&
+        !identifier.startsWith('no_stripe_customer_id')
+
+      if (isCustomer) {
+        try {
+          const existingCustomer = (await stripe.customers.retrieve(
+            identifier,
+          )) as Stripe.Response<Stripe.Customer>
+
+          await stripe.customers.update(identifier, {
+            email: targetUser.email,
+            name: targetUser.name || existingCustomer.name || targetUser.email,
+          })
+        } catch (error) {
+          console.error('Error updating stripe customer', error)
+        }
+      }
```

Comment:
this was the culprit since the inbound transfer had newer products but was also associated freshly with that shared MerchantCustomer row. This triggered the stripe webhook and BLAMMMMMO 💥